airflow/models/dag.py

```diff
@@ -1254,9 +1253,11 @@ def run(
         """
         from airflow.jobs import BackfillJob
         if not executor and local:
+            from airflow.executors.local_executor import LocalExecutor
             executor = LocalExecutor()
         elif not executor:
-            executor = get_default_executor()
+            from airflow.executors.executor_loader import ExecutorLoader
```

Comment:
(Not for this PR, but I do wonder if we should remove this functionality as it feels somewhat out of place with the "normal" way that `airflow tasks run` is used. It's almost closer to `airflow tasks submit` in nature. But it also bypasses lots of the scheduler "gating" that maybe it shouldn't.)

Reply:
I also do not feel confident about having pluginnable executors. I was quite surprised to see that it was possible.